pkg/authenticator/authenticator.go

```diff
@@ -242,40 +243,50 @@ func (auth *Authenticator) Authenticate() ([]byte, error) {
 		auth.Config.Username.FullUsername,
 	)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	resp, err := client.Do(req)
 	if err != nil {
-		return nil, log.RecordedError(log.CAKC027, err)
+		return log.RecordedError(log.CAKC027, err)
+	}
+
+	validatedResponse, err := ValidatedResponse(resp)
+	if err != nil {
+		return err
 	}
 
-	return DataResponse(resp)
+	parsedResponse, err := auth.parseAuthenticationResponse(validatedResponse)
```

Comment:
I wonder if it would help with readability to move everything in this function (Authenticate) above this line to a function `makeAuthenticationRequest`. 

This would then leave `Authenticate` looking something like this, where the `...` is logging and error handling:
```
func (auth *Authenticator) Authenticate() error {
    ...
    auth.makeAuthenticationRequest
    ...
    auth.parseAuthenticationResponse
    ...
    auth.AccessToken.Write
    ...
}
```